src/client/core/storage/storage-blob.service.ts

```diff
@@ -0,0 +1,359 @@
+    properties: {
+        contentLength: number;
+        contentType: string;
+        creationTime: Date | string | null;
+        lastModified: Date | string | null;
+    };
+}
+
+interface StorageBlobItem {
+    name: string;
+    url: string;
+    isDirectory: boolean;
+    properties?: {
+        contentLength?: number;
+        contentType: string;
+        creationTime: Date | string | null;
+        lastModified: Date | string | null;
+    }
+}
+
+type ContainerProperties = ContainerGetPropertiesResponse
+
+export interface ListBlobOptions {
+    /**
+     * Filter for the path.(Relative to the prefix if given)
+     */
+    folder?: string;
+
+    /**
+     * If it should list all files or 1 directory deep.
+     */
+    recursive?: boolean;
+
+    limit?: number;
+}
+
+const storageIpc = IpcEvent.storageBlob;
+
+@Injectable({ providedIn: "root" })
+export class StorageBlobService {
+    constructor(private ipcMain: BlIpcMain) {
+    }
+
+    public init() {
+        // TODO: Replace with method decorators?
+        this.ipcMain.on(storageIpc.listContainers, this.listContainers);
+        this.ipcMain.on(storageIpc.getContainerProperties,
+            this.getContainerProperties);
+        this.ipcMain.on(storageIpc.deleteContainer, this.deleteContainer);
+        this.ipcMain.on(storageIpc.createContainer, this.createContainer);
+
+        this.ipcMain.on(storageIpc.listBlobs, this.listBlobs);
+        this.ipcMain.on(storageIpc.getBlobProperties, this.getBlobProperties);
+        this.ipcMain.on(storageIpc.getBlobContent, this.getBlobContent);
+        this.ipcMain.on(storageIpc.generateSasUrl, this.generateSasUrl);
+        this.ipcMain.on(storageIpc.downloadBlob, this.downloadBlob);
+        this.ipcMain.on(storageIpc.uploadFile, this.uploadFile);
+        this.ipcMain.on(storageIpc.deleteBlob, this.deleteBlob);
+    }
+
+    @autobind()
+    public async listContainers(params: StorageBlobListContainersParams):
+        Promise<StorageBlobListContainersResult> {
+        const { prefix, continuationToken } = params;
+        const client = this.getServiceClient(params);
+        const containers = [];
+
+        // byPage() handles a null token as an actual token and returns 0 pages
+        let cToken = continuationToken ?? undefined;
+
+        const pages = client.listContainers({ prefix })
+            .byPage({ maxPageSize: 20, continuationToken: cToken });
+        for await (const page of pages) {
+            for (const container of page.containerItems) {
+                containers.push({
+                    ...container.properties,
+                    metadata: container.metadata,
+                    version: container.version,
+                    deleted: container.deleted,
+                    id: container.name
+                });
+            }
+            cToken = page.continuationToken;
+        }
+        return {
+            data: containers,
+            continuationToken: cToken
```

Comment:
If this did iterate over all of the pages we shouldn't return a continuation token. If something else is consuming this token we shouldn't be iterating over all the pages.

Reply:
As far as I can tell, at one point there were limits that were passed in, but they're no longer used. There's a `limit` parameter that was passed in from the `FileNavigator` to control the response page size, but it doesn't restrict how many pages are returned, so it's essentially a performance heuristic.

The `continuationToken` is handled gracefully by our code, such that if you return a partial response with a token, you can resume SDK calls from that point. It doesn't seem to have a rendering benefit, because the list is only rendered when all the items are returned. Maybe if there's some user-managed paging of blobs and containers, this could help, but I don't think we have any components that do that.

Based on some performance measures, I set some newer default page sizes for containers and blobs, and added an optional `maxPages` option, so that in some future time we can use it for paging or infinite scrolling. Also added some unit tests to show this behavior.

Lastly, I thought about handling a hard limit on the number of items, but it's not easy to implement, unless we start caching responses or require some other arguments. The `continuationToken` is opaque and lacks the granularity to resume mid-page. It can only be used to resume at the start of pages. Anyhow, probably not worth worrying about.